Sources/ParseSwift/Storage/PrimitiveObjectStore.swift

```diff
@@ -39,15 +44,27 @@ struct CodableInMemoryPrimitiveObjectStore: PrimitiveObjectStore {
 
 // MARK: KeychainStore + PrimitiveObjectStore
 extension KeychainStore: PrimitiveObjectStore {
+
     func delete(valueFor key: String) throws {
-        _ = removeObject(forKey: key)
+        if !removeObject(forKey: key) {
+            throw ParseError(code: .objectNotFound, message: "Object for key \"\(key)\" not found in Keychain")
```

Comment:
Do we need to throw here? IMO, if someone wants to delete something that doesn't exist, it's not a big deal. If the delete "succeeds" anyway, the net outcome is the same: the thing they're deleting doesn't exist.

Reply:
By the way, good catch on the rest of the error handling; I have no clue how I missed that.

Reply:
I believe I added the throw just because it conformed. I thought more of it if there was a problem deleting and object to throw.

Reply:
That makes sense; I forgot all the Keychain methods just return `Bool`.